sagemaker-neo-notebooks/edge/windows/tutorial.cc

```diff
@@ -0,0 +1,654 @@
+{
+    Aws::String s3_bucket_name(bucket_name.c_str(), bucket_name.size());
+    Aws::String s3_aws_object_name(model_name.c_str(), model_name.size());
+    Aws::S3::S3Client s3_client = getS3Client();
+
+    // upload model to s3 bucket
+    Aws::S3::Model::PutObjectRequest object_request;
+    object_request.SetBucket(s3_bucket_name);
+    object_request.SetKey(s3_aws_object_name);
+
+    // TODO verify this
+    const shared_ptr<Aws::IOStream> input_data =
+        Aws::MakeShared<Aws::FStream>(
+            "SampleAllocationTag",
+            filename.c_str(), ios_base::in | ios_base::binary);
+    object_request.SetBody(input_data);
+
+    // put model to the s3 bucket
+    auto put_object_outcome = s3_client.PutObject(object_request);
+    if (!put_object_outcome.IsSuccess())
+    {
+        auto error = put_object_outcome.GetError();
+        const std::string error_str = getErrorMessage(error);
+        throw std::runtime_error("UploadModel error: " + error_str);
+    }
+}
+
+bool checkBucketExist(string bucket_name)
+{
+    Aws::String s3_bucket_name(bucket_name.c_str(), bucket_name.size());
+    Aws::S3::S3Client s3_client = getS3Client();
+    auto list_bucket_resp = s3_client.ListBuckets();
+    if (list_bucket_resp.IsSuccess())
+    {
+        Aws::Vector<Aws::S3::Model::Bucket> bucket_list = list_bucket_resp.GetResult().GetBuckets();
+        for (auto const &bucket : bucket_list)
+        {
+            if (bucket.GetName().compare(s3_bucket_name) == 0)
+            {
+                return true;
+            }
+        }
+    }
+    else
+    {
+        auto error = list_bucket_resp.GetError();
+        const std::string error_str = getErrorMessage(error);
+        throw std::runtime_error("ListBucket error: " + error_str);
+    }
+    return false;
+}
+
+bool checkModelExist(string bucket_name, string model_name)
+{
+    Aws::String aws_bucket_name(bucket_name.c_str(), bucket_name.size());
+    Aws::String aws_model_name(model_name.c_str(), model_name.size());
+
+    Aws::S3::S3Client s3_client = getS3Client();
+    Aws::S3::Model::ListObjectsRequest objects_request;
+    objects_request.WithBucket(aws_bucket_name);
+
+    auto list_objects_outcome = s3_client.ListObjects(objects_request);
+    if (list_objects_outcome.IsSuccess())
+    {
+        Aws::Vector<Aws::S3::Model::Object> object_list = list_objects_outcome.GetResult().GetContents();
+        for (auto const &s3_object : object_list)
+        {
+            if (s3_object.GetKey().compare(aws_model_name) == 0)
+            {
+                return true;
+            }
+        }
+    }
+    else
+    {
+        auto error = list_objects_outcome.GetError();
+        const std::string error_str = getErrorMessage(error);
+        throw std::runtime_error("ListObjects error: " + error_str);
+    }
+
+    return false;
+}
+
+void UploadModelToS3(string model_name, string filename)
+{
+    const string s3_bucket_name = "windows-demo";
```

Comment:
I think the s3_bucket_name should not be hard-coded here and need to be passed by argument. The user should copy the model to their own bucket.

Reply:
it's revised.